include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -239,11 +239,11 @@ class kmer_hash_view<urng_t>::shape_iterator
     //!\brief Reference to `value_type`.
     using reference = value_type;
     //!\brief Tag this class as input iterator.
-    using iterator_category = typename std::iterator_traits<it_t>::iterator_category;
+    using iterator_category = detail::iterator_category_tag_t<it_t>;
     //!\brief Tag this class depending on which concept `it_t` models.
     using iterator_concept = std::conditional_t<std::contiguous_iterator<it_t>,
-                                                 typename std::random_access_iterator_tag,
-                                                 iterator_tag_t<it_t>>;
+                                                typename std::random_access_iterator_tag,
+                                                detail::iterator_concept_tag_t<it_t>>;
```

Comment:
This is the bit that fixes the issue, right?
Because no we do not use the input iterator tag, but the concept.

Reply:
`iterator_concept_tag_t` will return an iterator_tag, but based on the concept.

For example: If `it_t` is a `std::random_access_iterator`, `detail::iterator_concept_tag_t<it_t>` would return `std::random_access_iterator_tag`

Reply:
Yes, and before we just propagated the `iterator_category` which was wrong.